File: WebApi/1. Models/_TrackedModel.cs
namespace WebApi.Models
{
    public abstract class TrackedModel
    {
        /// <summary></summary>
        public DateTimeOffset? CreatedDate { get; set; }

        /// <summary></summary>
        public string? CreatedUserIp { get; set; }

        /// <summary></summary>
        public DateTimeOffset? UpdatedDate { get; set; }

        /// <summary></summary>
        public string? UpdatedUserIp { get; set; }
    }
}

File: WebApi/5. ClientPages/src/models/_tracked-model.ts
import { Type } from "class-transformer";

export abstract class TrackedModel
{
  /**  */
  @Type(() => Date)
  public createdDate?: Date;

  /**  */
  public createdUserIp: string;

  /**  */
  @Type(() => Date)
  public updatedDate?: Date;

  /**  */
  public updatedUserIp: string;
}
